torch_xla/csrc/tensor_util.cpp

```diff
@@ -904,6 +904,70 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
   }
 }
 
+std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
+    const std::vector<at::Tensor>& tensors,
+    const std::vector<XLATensor::ShardingSpecPtr>& shardings,
+    const std::vector<std::string>& devices) {
+  XLA_TIMED("TensorToData");
+  XLA_CHECK_EQ(tensors.size(), shardings.size());
+  XLA_CHECK_EQ(tensors.size(), devices.size());
+
+  std::vector<xla::ComputationClient::DataPtr> handles;
+  for (size_t i = 0; i < tensors.size(); ++i) {
+    torch::lazy::BackendDevice device = ParseDeviceString(devices[i]);
+    xla::Shape shape = CreateComputationShapeFromTensor(tensors[i], &device);
+
+    std::vector<xla::ComputationClient::TensorSource> source_tensors;  // in
+    std::vector<xla::ComputationClient::DataPtr> new_handles;          // out
+    if (shardings[i] != nullptr) {
```

Comment:
I think we should just assert on `shardings`. If there is no sharding, this function should not be called. User can user the other helper function

Reply:
We should also handle the implicitly replicated cases here, in which case the sharding[i] is null. I think it's easier to understand to handle these two cases in one place, as they are both related to SPMD mode. I could consider refactoring this a little bit, in the next PR?